source/Octopus.Tentacle.Client/Retries/RpcCallRetryHandler.cs

```diff
@@ -81,7 +81,34 @@ async Task<T> ExecuteAction(CancellationToken ct)
                     .WithRetryTimeout(remainingRetryDuration, timeoutStrategy)
                     .BuildTimeoutPolicy();
 
-                return await timeoutPolicy.ExecuteAsync(action, ct).ConfigureAwait(false);
+                if (timeoutStrategy == TimeoutStrategy.Optimistic)
+                {
+                    return await timeoutPolicy.ExecuteAsync(action, ct).ConfigureAwait(false);
+                }
+                else
+                {
```

Comment:
@nathanwoctopusdeploy would something like this work? or does the timeoutPolicy need to see an exception? The retry policy should still see an exception.